velox/exec/tests/utils/PlanBuilder.h

```diff
@@ -177,6 +181,7 @@ class PlanBuilder {
     }
 
     /// @param outputType List of column names and types to read from the table.
+    /// This property is required.
```

Comment:
Should we change the API to require outputType to be passed to startTableScan() ?

Reply:
There seems to be no clear consensus on the internet on if it's better to have mandatory builder arguments in the constructor or to throw an exception in the build method.

In this particular case I feel it's better to maintain the current API because:
- The `startTableScan()` function is currently used in 76 places across 9 different files. All these calls would have to updated to match the new API.
- I think the builder pattern of specifying arguments is a bit easier to read than having arguments in constructors.

Reply:
That said, @mbasmanova if you'd like to see what that change would look like, I have it on this branch:
https://github.com/ashkrisk/velox/compare/main...ashkrisk:velox:tbscan-alt-api

Reply:
@ashkrisk Thank you for clarifying your thinking. 

> The startTableScan() function is currently used in 76 places across 9 different files. All these calls would have to updated to match the new API.

I would not consider this point in making a decision. A one time refactor is a small cost to pay if it allows for a clear API. 

> Out of the two below snippets I think the second is a bit easier to read the first

I don't have a strong opinion. Let's just update startTableScan() documentation to clearly state that it must be followed by . outputType(type).